lib/task_bunny/message.ex

```diff
@@ -1,9 +1,42 @@
 defmodule TaskBunny.Message do
   @moduledoc """
-  Provides functionalities to access a message and its meta data.
+  Functions to access messages and its meta data.
   """
 
   @doc """
+  Encode message body in JSON with job and arugment.
+  """
+  @spec encode(atom, any) :: String.t
+  def encode(job, payload) do
+    %{
+      "job" => Atom.to_string(job),
+      "payload" => payload,
+      "created_at" => DateTime.utc_now()
+    }
+    |> Poison.encode!
+  end
+
+  @doc """
+  Decode message body in JSON to map
+  """
+  @spec decode(String.t) :: map
+  def decode(message) do
+    case Poison.decode(message) do
+      {:ok, decoded} ->
+        job = String.to_atom(decoded["job"])
```

Comment:
This won't work for module names. Module names will turn into these things:
```elixir
:"Jobs.ExampleJob"
```
which does not equal `Jobs.ExampleJob`

Check out how I solved this in Wobserver: https://github.com/shinyscorpion/wobserver/blob/master/lib/wobserver/util/helper.ex#L30

Btw: if you're trying to match an existing atom it is better to use: `String.to_existing_atom`. (Otherwise you might get memory leaks since atoms don't get garbage collected.)